src/unix/linux.c

```diff
@@ -1389,12 +1389,21 @@ static void uv__get_cgroup1_memory_limits(char buf[static 1024], uint64_t* high,
      * as indicated by uv__read_uint64 returning 0.
      */
      if (*high != 0 && *max != 0)
-       return;
+       goto update_limits;
   }
 
   /* Fall back to the limits of the global memory controller. */
   *high = uv__read_uint64("/sys/fs/cgroup/memory/memory.soft_limit_in_bytes");
   *max = uv__read_uint64("/sys/fs/cgroup/memory/memory.limit_in_bytes");
+
+  /* uv__read_uint64 detects cgroup2's "max", so we need to separately detect
+   * cgroup1's maximum value (which is derived from LONG_MAX and PAGE_SIZE).
+   */
+update_limits:
+  if (*high == LONG_MAX / 4096 * 4096)
```

Comment:
these seem like an odd magic constant, do you mean to do a bit mask here (LONG_MAX & ~4095)? and 4096 is not always page size (it is a kernel config flag)

Reply:
Updated to do bit twiddling instead (why is this preferred over the integer division/multiplication for rounding?), and using `sysconf` to query the page size.